libraries/core/classes/AppCommandRequest/index.js

```diff
@@ -0,0 +1,187 @@
+    return this;
+  }
+
+  /**
+   * Cleans up the instance after a successful or failed request.
+   * @private
+   * @param {Function} requestCallback The event callback that processes the response.
+   * @return {AppCommandRequest}
+   */
+  cleanUpRequest(requestCallback) {
+    // Remove the event listener entry.
+    event.removeCallback(this.getEventCallbackName(), requestCallback);
+    // Remove the request from the buffer.
+    requestBuffer.remove(this.serial);
+    return this;
+  }
+
+  /**
+   * Validates the command params before dispatch.
+   * @param {Object} commandParams The params of the command to be dispatched
+   * @return {boolean}
+   */
+  validateCommandParams() {
+    return true;
+  }
+
+  /**
+   * Creates a title for the app command request log
+   * @returns {string}
+   */
+  getRequestLogTitle() {
+    return `AppCommandRequest %c${this.commandName}`;
+  }
+
+  /**
+   * Creates a title for the app command response log
+   * @returns {string}
+   */
+  getResponseLogTitle() {
+    return `AppCommandResponse %c${this.commandName}`;
+  }
+
+  /**
+   * Creates payload for the app command response log.
+   *
+   * This default handler expects the command serial as first and the request response as
+   * second command response event parameter. If the class is used for request commands with a
+   * different response even payload, this method needs to be overwritten.
+   * @param {string} serial The serial that was used to identify the response callback.
+   * @param {Object} response The response object for the request
+   * @returns {Object}
+   */
+  getResponseLogPayload(serial, response) {
+    return response;
+  }
+
+  /**
+   * App command response handler. Can be used to implement custom logic for handling the incoming
+   * data.
+   *
+   * This default handler expects the command serial as first and the request response as
+   * second command response event parameter. If the class is used for request commands with a
+   * different response even payload, this method needs to be overwritten.
+   * @param {Function} resolve Resolve callback of the promise returned by the dispatch method
+   * of the AppCommandRequest instance.
+   * @param {Function} reject Reject callback of the promise returned by the dispatch method
+   * of the AppCommandRequest instance.
+   * @param {string} serial The serial that was used to identify the response callback.
+   * @param {Object} response The response object for the request
+   */
+  handleResponse(resolve, reject, serial, response) {
+    resolve(response);
+  }
+
+  /**
+   * Dispatches the app request command.
+   * @private
+   * @param {Function} resolve The resolve() callback of the dispatch promise.
+   * @param {Function} reject The reject() callback of the dispatch promise.
+   */
+  async onDispatch(resolve, reject) {
+    // Validate the command parameters.
+    if (this.validateCommandParams(this.commandParams) === false) {
```

Comment:
the "local" `validateCommandParams` method will also never return `false`. this method is supposed to be overwritten by classes that extend `AppCommandRequest`. In that case it might make sense to call it with `this.commandParams`.